applications/nrf_desktop/README.rst

```diff
@@ -1766,53 +1776,109 @@ The B0 bootloader requires the following options enabled:
+
+For an example of bootloader Kconfig configuration file defined by the application, see the MCUboot bootloader debug configuration defined for nRF52833 dongle (:file:`applications/nrf_desktop/configuration/nrf52833dongle_nrf52833/child_image/mcuboot/prj.conf`).
+
+.. note::
+  The nRF Desktop devices use either the serial recovery DFU with a single application slot or the background DFU.
+  Both mentioned firmware upgrade methods are not used simultaneously by any of the configurations.
+  For example, the ``nrf52840dk_nrf52840`` board in ``prj_mcuboot_smp.conf`` uses only background DFU and does not enable the serial recovery feature.
 
 .. _nrf_desktop_bootloader_background_dfu:
 
 Background Device Firmware Upgrade
 ==================================
 
 The nRF Desktop application uses the :ref:`nrf_desktop_config_channel` and :ref:`nrf_desktop_dfu` for the background DFU process.
+From the application perspective, the update image transfer during the background DFU process is very similar for both MCUboot and B0 bootloader.
 The firmware update process has three stages, discussed below.
 At the end of these three stages, the nRF Desktop application will be rebooted with the new firmware package installed.
 
 .. note::
-    The background DFU mode requires two application slots in the flash memory.
-    For this reason, the feature is not available for devices with smaller flash size, because the size of the flash memory required is essentially doubled.
-    The devices with smaller flash size can use either :ref:`nrf_desktop_bootloader_serial_dfu` or MCUboot bootloader with the secondary image partition located on an external flash.
-
-The background firmware upgrade can also be performed over the Simple Management Protocol (SMP).
-For more detailed information about the DFU over SMP, read the :ref:`nrf_desktop_ble_smp` documentation.
+  The background firmware upgrade can also be performed over the Simple Management Protocol (SMP).
+  For more detailed information about the DFU over SMP, read the :ref:`nrf_desktop_ble_smp` documentation.
 
 Update image generation
 -----------------------
 
-The update image is generated in the build directory when building the firmware if the bootloader is enabled in the configuration:
-
-* The :file:`zephyr/dfu_application.zip` file is used by both B0 and MCUboot bootloader for the background DFU through the :ref:`nrf_desktop_config_channel` and :ref:`nrf_desktop_dfu`.
-  This package contains firmware images along with additional metadata.
-
-  .. note::
-      By default, the build process for the B0 bootloader will construct an image for the first slot (slot 0 or S0).
-      To ensure that application is built for both slots, select the :kconfig:option:`CONFIG_BUILD_S1_VARIANT` Kconfig option.
+The update image is generated in the build directory when building the firmware when the bootloader is enabled in the configuration.
```

Comment:
```suggestion
The update image is generated in the build directory when building the firmware with the enabled bootloader.
```
or 'with the bootloader enabled in the configuration' if the latter part is needed